types/react/index.d.ts

```diff
@@ -905,51 +906,46 @@ declare namespace React {
     //
     // Event System
     // ----------------------------------------------------------------------
-
-    interface SyntheticEvent<T = Element> {
+    interface BaseSyntheticEvent<E = any, T = any, C = any> {
+        nativeEvent: E;
+        target: T;
+        currentTarget: C;
         bubbles: boolean;
-        /**
-         * A reference to the element on which the event listener is registered.
-         */
-        currentTarget: EventTarget & T;
```

Comment:
I'm concerned that we are changing the signature of `SyntheticEvent` when its not needed but I doubt anyone actually referenced it, so its probably okay.

Reply:
Interestingly it seems that the specific events always had their `nativeEvent` unioned with the generic `Event` interface, but now that's different so it is a breaking change I think for those properties.

Reply:
@ericanderson I originally wrote a `DOMSyntheticEvent` interface that all of the DOM events inherited from and wrote `SyntheticEvent` to keep its current signature. Something like this:

```ts
interface DOMSyntheticEvent<T = Element, E = Event> extends BaseSyntheticEvent<
  E,
  EventTarget,
  EventTarget & T
> {}

type SyntheticEvent<T = Element> = DOMSyntheticEvent<T, Event>
```
I changed it to the current state because that would be breaking for anyone that uses `SyntheticEvent` as a generic interface for DOM events, correct?

@ferdaber I'm not positive that I follow. Are you saying that this:
```ts
interface A {
  one: Foo;
}

interface B extends A {
  one: Bar;
}
```
is different than this:
```ts
interface A<T = Foo> {
  one: T;
}

interface B extends A<Bar> {}
```
In my mind those have the same result, but I may very well be missing something.

Reply:
Oh that's my mistake, I didn't read it closely enough, since the specific events are extending from `SyntheticEvent` the `nativeEvent` properties are proper subtypes of the `Event` interface so they're not unioned. So we should be good on my end.

Reply:
Yeah I wont block this.

Reply:
@pshrmn If I'm not mistaken, the first option will trigger an error since you cannot override the types of inherited interfaces.